File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/patch_session_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class PatchSessionDetails(object):
    """
    Details to update a session.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new PatchSessionDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param id:
            The value to assign to the id property of this PatchSessionDetails.
        :type id: str

        :param name:
            The value to assign to the name property of this PatchSessionDetails.
        :type name: str

        :param path:
            The value to assign to the path property of this PatchSessionDetails.
        :type path: str

        :param type:
            The value to assign to the type property of this PatchSessionDetails.
        :type type: str

        :param kernel:
            The value to assign to the kernel property of this PatchSessionDetails.
        :type kernel: oci.aidataplatform_dp.models.Kernel

        :param cluster_id:
            The value to assign to the cluster_id property of this PatchSessionDetails.
        :type cluster_id: str

        :param must_refresh_context:
            The value to assign to the must_refresh_context property of this PatchSessionDetails.
        :type must_refresh_context: bool

        """
        self.swagger_types = {
            'id': 'str',
            'name': 'str',
            'path': 'str',
            'type': 'str',
            'kernel': 'Kernel',
            'cluster_id': 'str',
            'must_refresh_context': 'bool'
        }

        self.attribute_map = {
            'id': 'id',
            'name': 'name',
            'path': 'path',
            'type': 'type',
            'kernel': 'kernel',
            'cluster_id': 'cluster_id',
            'must_refresh_context': 'must_refresh_context'
        }

        self._id = None
        self._name = None
        self._path = None
        self._type = None
        self._kernel = None
        self._cluster_id = None
        self._must_refresh_context = None

    @property
    def id(self):
        """
        Gets the id of this PatchSessionDetails.
        UUID of the session.


        :return: The id of this PatchSessionDetails.
        :rtype: str
        """
        return self._id

    @id.setter
    def id(self, id):
        """
        Sets the id of this PatchSessionDetails.
        UUID of the session.


        :param id: The id of this PatchSessionDetails.
        :type: str
        """
        self._id = id

    @property
    def name(self):
        """
        Gets the name of this PatchSessionDetails.
        A user-friendly name for the session.


        :return: The name of this PatchSessionDetails.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this PatchSessionDetails.
        A user-friendly name for the session.


        :param name: The name of this PatchSessionDetails.
        :type: str
        """
        self._name = name

    @property
    def path(self):
        """
        Gets the path of this PatchSessionDetails.
        Path to the session. A directory where notebook server is started and where notebooks are saved. For example, /data/test.ipynb.


        :return: The path of this PatchSessionDetails.
        :rtype: str
        """
        return self._path

    @path.setter
    def path(self, path):
        """
        Sets the path of this PatchSessionDetails.
        Path to the session. A directory where notebook server is started and where notebooks are saved. For example, /data/test.ipynb.


        :param path: The path of this PatchSessionDetails.
        :type: str
        """
        self._path = path

    @property
    def type(self):
        """
        Gets the type of this PatchSessionDetails.
        Type of session.


        :return: The type of this PatchSessionDetails.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this PatchSessionDetails.
        Type of session.


        :param type: The type of this PatchSessionDetails.
        :type: str
        """
        self._type = type

    @property
    def kernel(self):
        """
        Gets the kernel of this PatchSessionDetails.

        :return: The kernel of this PatchSessionDetails.
        :rtype: oci.aidataplatform_dp.models.Kernel
        """
        return self._kernel

    @kernel.setter
    def kernel(self, kernel):
        """
        Sets the kernel of this PatchSessionDetails.

        :param kernel: The kernel of this PatchSessionDetails.
        :type: oci.aidataplatform_dp.models.Kernel
        """
        self._kernel = kernel

    @property
    def cluster_id(self):
        """
        Gets the cluster_id of this PatchSessionDetails.
        Cluster ID.


        :return: The cluster_id of this PatchSessionDetails.
        :rtype: str
        """
        return self._cluster_id

    @cluster_id.setter
    def cluster_id(self, cluster_id):
        """
        Sets the cluster_id of this PatchSessionDetails.
        Cluster ID.


        :param cluster_id: The cluster_id of this PatchSessionDetails.
        :type: str
        """
        self._cluster_id = cluster_id

    @property
    def must_refresh_context(self):
        """
        Gets the must_refresh_context of this PatchSessionDetails.
        Set as True, if a new execution context is needed.


        :return: The must_refresh_context of this PatchSessionDetails.
        :rtype: bool
        """
        return self._must_refresh_context

    @must_refresh_context.setter
    def must_refresh_context(self, must_refresh_context):
        """
        Sets the must_refresh_context of this PatchSessionDetails.
        Set as True, if a new execution context is needed.


        :param must_refresh_context: The must_refresh_context of this PatchSessionDetails.
        :type: bool
        """
        self._must_refresh_context = must_refresh_context

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/PatchSessionDetails.java
package com.oracle.aidataplatform.dp.model;



/**
 * Details to update a session.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=PatchSessionDetails.Builder.class)

public final class PatchSessionDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"id", "name", "path", "type", "kernel", "clusterId", "mustRefreshContext"})
    public PatchSessionDetails(String id, String name, String path, Session.Type type, Kernel kernel, String clusterId, Boolean mustRefreshContext) {
        super();
        this.id = id;
        this.name = name;
        this.path = path;
        this.type = type;
        this.kernel = kernel;
        this.clusterId = clusterId;
        this.mustRefreshContext = mustRefreshContext;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * UUID of the session.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("id")
private String id;

        /**
         * UUID of the session.
         * @param id the value to set
         * @return this builder
         **/
        

public Builder id(String id) {
    this.id = id;
    return this;
}
            /**
     * A user-friendly name for the session.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;

        /**
         * A user-friendly name for the session.
         * @param name the value to set
         * @return this builder
         **/
        

public Builder name(String name) {
    this.name = name;
    return this;
}
            /**
     * Path to the session. A directory where notebook server is started and where notebooks are saved. For example, /data/test.ipynb.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("path")
private String path;

        /**
         * Path to the session. A directory where notebook server is started and where notebooks are saved. For example, /data/test.ipynb.
         * @param path the value to set
         * @return this builder
         **/
        

public Builder path(String path) {
    this.path = path;
    return this;
}
            /**
     * Type of session.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private Session.Type type;

        /**
         * Type of session.
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(Session.Type type) {
    this.type = type;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("kernel")
private Kernel kernel;



public Builder kernel(Kernel kernel) {
    this.kernel = kernel;
    return this;
}
            /**
     * Cluster ID.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("cluster_id")
private String clusterId;

        /**
         * Cluster ID.
         * @param clusterId the value to set
         * @return this builder
         **/
        

public Builder clusterId(String clusterId) {
    this.clusterId = clusterId;
    return this;
}
            /**
     * Set as True, if a new execution context is needed.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("must_refresh_context")
private Boolean mustRefreshContext;

        /**
         * Set as True, if a new execution context is needed.
         * @param mustRefreshContext the value to set
         * @return this builder
         **/
        

public Builder mustRefreshContext(Boolean mustRefreshContext) {
    this.mustRefreshContext = mustRefreshContext;
    return this;
}


        public PatchSessionDetails build() {
            PatchSessionDetails model = new PatchSessionDetails(this.id
                , this.name
                , this.path
                , this.type
                , this.kernel
                , this.clusterId
                , this.mustRefreshContext);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(PatchSessionDetails model) {
                this.id(model.getId());
    this.name(model.getName());
    this.path(model.getPath());
    this.type(model.getType());
    this.kernel(model.getKernel());
    this.clusterId(model.getClusterId());
    this.mustRefreshContext(model.getMustRefreshContext());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * UUID of the session.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("id")
    private final String id;

        /**
     * UUID of the session.
     * @return the value
     **/
    
    public String getId() {
        return id;
    }


        /**
     * A user-friendly name for the session.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * A user-friendly name for the session.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * Path to the session. A directory where notebook server is started and where notebooks are saved. For example, /data/test.ipynb.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("path")
    private final String path;

        /**
     * Path to the session. A directory where notebook server is started and where notebooks are saved. For example, /data/test.ipynb.
     * @return the value
     **/
    
    public String getPath() {
        return path;
    }


        /**
     * Type of session.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final Session.Type type;

        /**
     * Type of session.
     * @return the value
     **/
    
    public Session.Type getType() {
        return type;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("kernel")
    private final Kernel kernel;

    
    public Kernel getKernel() {
        return kernel;
    }


        /**
     * Cluster ID.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("cluster_id")
    private final String clusterId;

        /**
     * Cluster ID.
     * @return the value
     **/
    
    public String getClusterId() {
        return clusterId;
    }


        /**
     * Set as True, if a new execution context is needed.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("must_refresh_context")
    private final Boolean mustRefreshContext;

        /**
     * Set as True, if a new execution context is needed.
     * @return the value
     **/
    
    public Boolean getMustRefreshContext() {
        return mustRefreshContext;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("PatchSessionDetails(");
        sb.append("id=").append(String.valueOf(this.id));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", path=").append(String.valueOf(this.path));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", kernel=").append(String.valueOf(this.kernel));
        sb.append(", clusterId=").append(String.valueOf(this.clusterId));
        sb.append(", mustRefreshContext=").append(String.valueOf(this.mustRefreshContext));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof PatchSessionDetails)) {
            return false;
        }

        PatchSessionDetails other = (PatchSessionDetails) o;
        return java.util.Objects.equals(this.id, other.id) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.path, other.path) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.kernel, other.kernel) &&
            java.util.Objects.equals(this.clusterId, other.clusterId) &&
            java.util.Objects.equals(this.mustRefreshContext, other.mustRefreshContext);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.id == null ? 43 : this.id.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.path == null ? 43 : this.path.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.kernel == null ? 43 : this.kernel.hashCode());
        result = (result * PRIME) + (this.clusterId == null ? 43 : this.clusterId.hashCode());
        result = (result * PRIME) + (this.mustRefreshContext == null ? 43 : this.mustRefreshContext.hashCode());
        return result;
    }


}
